lib/sign.js

```diff
@@ -17,11 +18,51 @@ var xpi = require("./xpi");
 
 var AMO_API_PREFIX = require("./settings").AMO_API_PREFIX;
 
+function getXpiInfo(options) {
+  options = _.assign({
+    addonDir: null,
+    xpiPath: null,
+  }, options);
+
+  if (!options.xpiPath) {
+    return utils.getManifest({addonDir: options.addonDir})
+      .then(function(manifest) {
+        return getXpiInfoFromManifest(manifest);
+      });
+  } else {
+    return utils.extractXPI(options.xpiPath)
```

Comment:
It doesn't feel right to duplicate loading the package.json file but I guess it isn't much code. You could extract that into a method and make this just set `var manifest = utils.loadManifest(tmpXPI)`. It's small but it might start getting duplicated in lots of places.